src/patients/ContactInfo.tsx

```diff
@@ -0,0 +1,211 @@
+  name: string
+  isEditable?: boolean
+  onChange?: (newData: ContactInfoPiece[]) => void
+  type: 'text' | 'email' | 'tel'
+  isValid?: (entry: string) => boolean
+  errorMessageLabel: string
+}
+
+const ContactInfo = (props: Props) => {
+  const {
+    data,
+    errors,
+    label,
+    name,
+    isEditable,
+    onChange,
+    type,
+    isValid,
+    errorMessageLabel,
+  } = props
+  const [tempErrors, setTempErrors] = useState<string[]>()
+
+  const { t } = useTranslation()
+
+  useEffect(() => {
+    if (!onChange) {
+      return
+    }
+    if (data.length === 0) {
+      onChange([...data, { value: '' }])
+    }
+  }, [data, onChange])
+
+  const getError = (i: number) => {
+    const tempError = tempErrors ? tempErrors[i] : null
+    if (tempError) {
+      return t(tempError)
+    }
+
+    const error = errors ? errors[i] : null
+    if (error) {
+      return t(error)
+    }
+
+    return ''
+  }
+
+  const typeLabel = t('patient.contactInfoType.label')
+  const typeOptions = Object.values(ContactInfoTypes).map((value) => ({
+    label: t(`patient.contactInfoType.options.${value}`),
+    value: `${value}`,
+  }))
+
+  const addLabel = t('actions.add')
+
+  const onChangeValue = (event: any, prevValue: string) => {
+    if (!onChange) {
+      return
+    }
+
+    // eslint-disable-next-line no-shadow
+    const newData = data.map(({ value, type }) =>
+      value === prevValue ? { value: event.currentTarget.value, type } : { value, type },
+    )
+    onChange(newData)
+  }
+
+  const header = (
+    <Row className="header mb-2">
+      <Column xs={12} sm={4}>
+        <span className="">{typeLabel}</span>
+        <span className="d-sm-none"> &amp; {t(label)}</span>
+      </Column>
+      <Column className="d-none d-sm-block" sm={8}>
+        {t(label)}
+      </Column>
+    </Row>
+  )
+
+  // todo: acts strange when deleting empty rows above non-empty rows.
+  // suspect TextInputWithLabelFormGroup missing value
+  const entries = data.map((entry, i) => {
+    const error = getError(i)
+    return (
+      // todo: want a better key, and possibly name
+      // eslint-disable-next-line react/no-array-index-key
+      <Row key={i}>
+        <Column sm={4}>
+          <SelectWithLabelFormGroup
+            name={`${name}Type${i}`}
+            value={entry.type}
+            isEditable={isEditable}
+            options={typeOptions}
+            onChange={
+              onChange
+                ? (event) => {
+                    // eslint-disable-next-line no-shadow
+                    const newData = data.map(({ value, type }) =>
+                      value === entry.value
+                        ? { value, type: event.currentTarget.value }
+                        : { value, type },
+                    )
+                    onChange(newData)
+                  }
+                : undefined
+            }
+          />
+        </Column>
+        <Column sm={8}>
+          {['tel', 'email'].indexOf(type) > -1 ? (
```

Comment:
i think we can just pass in the type of text input box we wish to display as a prop to contact info component. 

to me it seems like it is less likely to break if we explicitly define it as a prop. Otherwise, the logic for determining whether or not the text box is a phone number or email is based on the types of the labels for the contact information types we support. 

it also has the added benefit of removing the need for the conditional.

Reply:
✅